Android/app/src/main/java/com/example/boardgamesocial/MainApp/Fragments/GameCollectionFragment.java

```diff
@@ -107,6 +126,43 @@ public void onDestroyView() {
 
     @Override
     public void onItemClick(Bundle contextBundle) {
-        Toast.makeText(getContext(),"Game clicked", Toast.LENGTH_LONG).show();
+        //Toast.makeText(getContext(),"Reviews functionality to be implemented!", Toast.LENGTH_LONG).show();
+
+        NavHostFragment.findNavController(GameCollectionFragment.this)
+                .navigate(R.id.action_gameCollectionFragment_to_gameReviews, contextBundle);
+    }
+
+    //Method to remove private games from the games collection
+    private Observable<JsonArray> fetchNonPrivateGames() {
+
+        Log.i(TAG, "Inside fetchNonPrivateGames");
+
+        gameToUserList = new ArrayList<>();
+
+        RetrofitClient.getClient().getCall(GameToUser.class, new HashMap<String, String>() {{
+            put("private", "True");
+        }}).blockingSubscribe(gameToUserJson -> gameToUserList = getObjectList(gameToUserJson, GameToUser.class));
+
+        List<Integer> gameIdList = gameToUserList.stream().map(GameToUser::getGameId).collect(Collectors.toList());
+
+        JsonArray jsonArray = new JsonArray();
+
+        RetrofitClient.getClient().getCall(Game.class, new HashMap<>()).blockingSubscribe(j -> {
+            if(!j.isEmpty()) {
+
+                List<Game> gameList = getObjectList(j, Game.class);
+
+                if(Objects.nonNull(gameList) && !gameList.isEmpty()) {
+
+                    gameList.forEach(game -> {
+                        if(!gameIdList.contains(game.getId())) {
+                            jsonArray.add(gson.toJsonTree(game));
+                        }
+                    });
+                }
+            }
+        });
+
+        return Observable.fromArray(jsonArray);
```

Comment:
It seems you are using jsonArray from mainThread in io thread for requests, that might cause issues with return being null, since it might take some time to receive response on request. What you probably should do is whatever action should be invoked on response should happen inside the subscriber.

Reply:
The current issue with the game collection is there is an issue with refreshing the page. I have tried RetrofitClient.getClient().getCall which does the refreshing correctly but then private games aren’t fetched. Though shall take a look into it and put the response inside the subscriber